### PR TITLE
#349 - added docs comments

### LINKED ISSUE #349: Add docs comments
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/pageres/issues/349)

Add JSDoc/TSDoc comments to all the methods and options, to improve the experience for TypeScript users. See: https://github.com/sindresorhus/typescript-definition-style-guide for how to do it. A lot of the text can be copied from the readme.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $20.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($20.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/pageres/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by gurrrung**
Fixes #349  Added docs TS/JS comments

### CODE REVIEW

**Comment by gurungrahul2** [comment]
Following the build log, removing those two lines from test, is causing further tests to fail.

**Comment by gurungrahul2** [comment]
Found same error in different PR build log , https://github.com/sindresorhus/pageres/pull/356 @sindresorhus

**Comment by gurungrahul2** [comment]
@sindresorhus

File: source/index.ts [hunk 1]
```diff
@@ -21,42 +21,159 @@ import unusedFilename from 'unused-filename';
...
+	 */
 	username?: string;
+
+	/**
+	 * Password for authenticating with HTTP auth.
+	 */
 	password?: string;
+
+	/**
+	 * Scale webpage number times.
```
Comment (sindresorhus) [inline source/index.ts]: This is not a valid sentence.

File: source/index.ts [hunk 2]
```diff
@@ -21,42 +21,159 @@ import unusedFilename from 'unused-filename';
...
 	sizes: string[];
+
+	/**
+	 * Options set here will take precedence over the ones set in the constructor.
+	 */
 	options?: Options;
 }
 
+/**
+* Set the destination directory.
```
Comment (sindresorhus) [inline source/index.ts]: This comment is not correct. The type *is* the destination directory.

File: source/index.ts [hunk 3]
```diff
@@ -21,42 +21,159 @@ import unusedFilename from 'unused-filename';
 const writeFile = promisify(fs.writeFile);
 
 export interface Options {
+	/**
+	 * Delay capturing the screenshot (seconds).
+	 *
+	 * @default 0
+	 */
```
Comment (sindresorhus) [inline source/index.ts]: I changed my comment style. See https://github.com/sindresorhus/typescript-definition-style-guide/commit/6a256ee82c9e8b32a764f60f8362fff1879a26ad

File: source/index.ts [hunk 4]
```diff
@@ -21,42 +21,159 @@ import unusedFilename from 'unused-filename';
...
 	script?: string;
+
+	/**
+	 * A string with the same format as a browser cookie or an object.
+	 */
 	cookies?: (string | {[key: string]: string})[];
+
+	/**
+	 * Define a customized filename using Lo-Dash templates.
```
Comment (sindresorhus) [inline source/index.ts]: The readme contains a lot more info that is missing here.

**Comment by sindresorhus** [comment]
Include all the text from the options in the readme, not just the first sentence. Also document the methods and include a code example.

**Comment by sindresorhus** [comment]
> Following the build log, removing those two lines from test, is causing further tests to fail.

Rebase from `master` branch. There are no failures there.

**Comment by gurungrahul2** [comment]
> Include all the text from the options in the readme, not just the first sentence. Also document the methods and include a code example.

how should I add type, for example, Type: `number` *(seconds)*<br>
should be written as same with <br> and backticks ? @sindresorhus

**Comment by sindresorhus** [comment]
It's documented in https://github.com/sindresorhus/typescript-definition-style-guide which is linked to from the issue.

**Comment by sindresorhus** [comment]
Ping

**Comment by gurrrung** [comment]
Can I continue this work too, please?

**Pull request closed without merging**